api/src/main/scala/hmda/api/http/ValidationErrorConverter.scala

```diff
@@ -62,13 +70,29 @@ trait ValidationErrorConverter {
       .getOrElse("")
   }
 
-  private def relevantFields(err: ValidationError): JsObject = {
+  private def relevantRow(err: ValidationError, vs: HmdaFileValidationState): HmdaFileRow = {
+    if (err.ts) vs.ts.getOrElse(HmdaRowError())
+    else vs.lars.find(lar => lar.loan.id == err.errorId).getOrElse(HmdaRowError())
+  }
+
+  private def relevantFields(err: ValidationError, row: HmdaFileRow): JsObject = {
     val fieldNames: Seq[String] = editDescriptions.find(e => e.editNumber == err.ruleName)
```

Comment:
While I realize that this comment refers to legacy, I figure it might be useful anyway since you are already deeply submerged in this part of the code.

We are performing the search `editDescriptions.find(e => e.editNumber == err.ruleName)`, though coded slightly differently, both here and in `findEditDescription` and using them to find the same object. It would probably be better to find that object separately and just extracted the needed pieces from it here and in  `findEditDescription`.